src/tools/pub/cmd/pub-add.md

```diff
@@ -26,65 +26,149 @@ For example, if 0.13.3 is the latest stable version of the `http` package,
 then `dart pub add http` adds
 `http: ^0.13.3` under `dependencies` in the pubspec.
 
-To add a [dev dependency][], use the `--dev` option:
-
-[dev dependency]: /tools/pub/dependencies#dev-dependencies
+You can also specify a constraint or constraint range
 
 ```terminal
-$ dart pub add --dev test
+$ dart pub add foo:2.0.0
+$ dart pub add foo:'^2.0.0'
+$ dart pub add foo:'>2.0.0 <3.0.1'
 ```
 
-## Options
+If `dart pub add <package>:<constraint>` is an existing dependency,
+it will update the constraint.
 
-For options that apply to all pub commands, see
-[Global options](/tools/pub/cmd#global-options).
+## YAML descriptor
 
-{{site.alert.note}}
-  Any specified options will apply to all the packages
-  included in an invocation of the command.
-  For example, `dart pub add test http --dev` 
-  will add both the `test` and `http` packages 
-  as dev dependencies.
+{{site.alert.version-note}}
+  YAML-formatted descriptor syntax was added in Dart 2.19.
+  The descriptor replaces arguments like `--path`, `--sdk`, `--git-<option>`, etc.
+  We still support these arguments, but the documented method is now YAML-descriptor only.
 {{site.alert.end}}
 
-### `-d, --dev`
+The YAML descriptor reflects how dependencies are written in `pubspec.yaml`.
+
+```nocode
+$ dart pub add [options] [dev:]<package>[:descriptor] [[dev:]<package>[:descriptor]
+       ...]
+```
+
+Within the `descriptor` you can add a package with specific constraints or other sources:
```

Comment:
Remember for here and elsewhere, try to stick to `<80` characters with semantic breaks: https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks

This helps for review in GitHub when writing, but especially when editing later.

I recommend adding a visual guideline to your editor for 80 characters ([IntelliJ family](https://www.jetbrains.com/help/idea/settings-code-style.html#code-style-setting-general:~:text=specified%20right%20margin.-,Visual%20guides,-In%20this%20field), [VSCode](https://code.visualstudio.com/docs/getstarted/settings#:~:text=is%20empty.%0A%20%20%22-,editor.rulers,-%22%3A%20%5B%5D%2C%0A%0A%20%20//%20Controls)).